store/models.py

```diff
@@ -9,10 +9,48 @@ class Book(models.Model):
     description = models.TextField(null=True)
     mrp = models.PositiveIntegerField()
     rating = models.FloatField(default=0.0)
-
+    ratings = []
     class Meta:
         ordering = ('title',)
+    def checkUser(self,user):
+        ind=0
+        for i in self.ratings:
+            if(i[0]==user):
```

Comment:
There's an error with your rating. You should have created another model here to store the Book, the User and the rating given by that user. And then to calculate the average rating of a book, you could have filtered that model to get the corresponding book and the user, and then calculated the average rating, and stored it in the rating field of the Book model.

Currently, when multiple users rate a book, the average rating of the book becomes equal to the last user who has rated it.

Reply:
No that is not the case i have used an array for storing them with the username and have checked it also

Reply:
Yeah, you've used array but the array isn't stored anywhere in the database. So, the array will not persist when you close the server and then re-start it. So, all the user ratings for a book would be gone, and then when a user will rate the book, it will be considered as if the rating is inserted in an empty array.

An array isn't persistent. There's no proper field like an array to store data in models. For that, we create another Model using some `ForeignKey` like relation.

As an example, check the admin panel, you won't find the array to be stored anywhere.